modules/foreman_host.py

```diff
@@ -0,0 +1,172 @@
+  - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.7"
+author:
+  - "Bernhard Hopfenmueller (@Fobhep) ATIX AG"
+requirements:
+  - "nailgun >= 0.29.0"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+
+# NEEDS REWORK
+
+
+'''
+
+EXAMPLES = '''
+- name: "Create a host"
+  foreman_host:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_host"
+    hostgroup: my_hostgroup
+    state: present
+
+- name: "Switch off an existing host"
+  foreman_host:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_host"
+    enabled: false
+    state: present
+
+- name: "Delete a host"
+  foreman_host:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "new_host"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+try:
+    from nailgun.entities import (
+        Host,
+        HostGroup,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        find_host,
+        find_hostgroup,
+        find_location,
+        find_organization,
+        find_parameter_from_hostgroup,
+        # set_power_state,
+        naildown_entity_state,
+        sanitize_entity_dict,
+    )
+except ImportError:
+    pass
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    # 'mac': 'mac',
+    'enabled': 'enabled',
+    'hostgroup': 'hostgroup',
+    'location': 'location',
+    'organization': 'organization',
+}
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            hostgroup=dict(),
+            location=dict(),
+            # mac=dict(default='00-00-00-00-00-00'),
+            organization=dict(),
+            # power_state=dict(default='on'),
+            enabled=dict(default='true', type='bool'),
+            state=dict(default='present', choices=[
+                       'present_with_defaults', 'present', 'absent']),
+        ),
+        required_if=(
+            ['state', 'present_with_defaults', ['hostgroup']],
+            ['state', 'present', ['hostgroup']],
+        ),
+        supports_check_mode=True,
+    )
+
+    (host_dict, state) = module.parse_params()
+
+    module.connect()
+    host_dict['hostgroup'] = find_hostgroup(
+        module, host_dict['hostgroup'], failsafe=True)
+    host_dict['name'] = host_dict['name'] + '.' + \
+        find_parameter_from_hostgroup(
+            module, host_dict['hostgroup'], 'domain_name')
```

Comment:
What is the name already includes the domain? We should probably have some logic here, if not at least document it

Reply:
I documented it - for now the name should not include the domain